### PR TITLE
Split image generation engine

### LINKED ISSUE #525: Split monolithic ImageGenerationEngine
ImageGenerationEngine currently defines four methods supporting four image generation modes:
- text to image
- image variation
- image outpainting
- image inpainting

Instead, define four independent engines to support each mode:
- PromptImageGenerationEngine
- VariationImageGenerationEngine
- InpaintingImageGenerationEngine
- OutpaintingImageGenerationEngine

### PR DESCRIPTION
**Pull request by andrewfrench**
Resolves #525 

This PR splits the monolithic image generation engine into one engine that handles each image generation mode:
- prompt (text to image) generation
- image variation
- inpainting
- outpainting

The monolithic image generation tool (ImageGenerator) has been split to match, now reflecting both the engine and the task structures.

A new mixin (ImageArtifactFileOutputMixin) has been added to handle writing image artifacts to files locally. This can be expanded later to handle media artifacts in general or, if desired, any artifact.

### CODE REVIEW

File: griptape/tools/__init__.py [hunk 1]
```diff
@@ -43,5 +46,8 @@
     "TaskMemoryClient",
     "Computer",
     "OpenWeatherClient",
-    "ImageGenerator",
+    "PromptImageGenerator",
+    "ImageVariationGenerator",
+    "ImageInpainter",
+    "ImageOutpainter",
```
Comment (collindutter) [inline griptape/tools/__init__.py:52]: Similar naming comment. And maybe we should rename to be "Clients"? `InpaintingImageGenerationClient`?

**Review by collindutter — COMMENTED** [review]
Really nice work, just some comments on naming standardization between Tasks, Engines, and Tools.

File: griptape/engines/__init__.py [hunk 2]
```diff
@@ -15,5 +19,9 @@
     "BaseExtractionEngine",
     "CsvExtractionEngine",
     "JsonExtractionEngine",
-    "ImageGenerationEngine",
+    "BaseImageGenerationEngine",
+    "PromptImageGenerationEngine",
+    "VariationImageGenerationEngine",
+    "InpaintingImageGenerationEngine",
+    "OutpaintingImageGenerationEngine",
```
[the remark this answers is not in the document]
Comment (andrewfrench) [reply griptape/engines/__init__.py:26]: I like that -- updated.

**Review by collindutter — APPROVED** [review]
Nice work 🤘

**Pull request merged**